batch7crud-roll5/src/main/java/com/lftechnology/batch7crud/dao/EmployeeDAO.java

```diff
@@ -0,0 +1,116 @@
+            ResultSet result = ps.executeQuery();
+
+            while (result.next()) {
+                emp = new Employee();
+                emp.setId(result.getInt("id"));
+                emp.setFirstName(result.getString("first_name"));
+                emp.setLastName(result.getString("last_name"));
+                emp.setDepartment(result.getString("department"));
+                emp.setAddress(result.getString("address"));
+                empList.add(emp);
+
+            }
+
+            return empList;
+
+        } catch (SQLException e) {
+            throw new DataException();
+        }
+
+    }
+
+    public Employee fetchById(int id) throws DataException {
+        try {
+            Employee emp = null;
+            String sql = "SELECT * FROM employee where id = ?";
+
+            Connection conn = DBConnection.getConnection();
+            PreparedStatement ps = conn.prepareStatement(sql);
+            ps.setInt(1, id);
+
+            ResultSet rs = ps.executeQuery();
+
+            while (rs.next()) {
+                emp = new Employee();
+                emp.setId(rs.getInt("id"));
+                emp.setFirstName(rs.getString("first_name"));
+                emp.setLastName(rs.getString("last_name"));
+                emp.setDepartment(rs.getString("department"));
+                emp.setAddress(rs.getString("address"));
+
+            }
+
+            return emp;
+
+        } catch (SQLException e) {
+            throw new DataException();
+        }
+
+    }
+
+    public boolean create(Employee employee) throws DataException {
+        try {
+            String sql = "insert into employee (first_name, last_name, department, address) values(?, ?, ?, ?)";
+            Connection conn = DBConnection.getConnection();
+            PreparedStatement ps = conn.prepareStatement(sql);
+
+            ps.setString(1, employee.getFirstName());
+            ps.setString(2, employee.getLastName());
+            ps.setString(3, employee.getDepartment());
+            ps.setString(4, employee.getAddress());
+
+            if (ps.executeUpdate() != 0) {
+                return true;
+            } else {
+                return false;
+            }
+
+        } catch (SQLException e) {
+            throw new DataException();
+        }
+
+    }
+
+    public void edit(Employee employee, int id) throws DataException {
+        try {
+            String sql = "update employee set first_name = ?, last_name = ?, department = ?, address = ? where id = ?";
+            Connection conn = DBConnection.getConnection();
+            PreparedStatement ps = conn.prepareStatement(sql);
+
+            ps.setString(1, employee.getFirstName());
+            ps.setString(2, employee.getLastName());
+            ps.setString(3, employee.getDepartment());
+            ps.setString(4, employee.getAddress());
+            ps.setInt(5, id);
+
+            ps.executeUpdate();
+
+        } catch (SQLException e) {
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) Either log or rethrow this exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1166)